--- BuggyCode.java (Com Erro)
+++ GroundTruth_Fix.java (Corrigido)
@@ -247,9 +247,12 @@
 								ConversationProvider.INDEX_THREADID]
 								+ " = " + threadId, null, null);
 				if (cursor != null && cursor.moveToFirst()) {
-					return getConversation(context, cursor, true);
+					ret = getConversation(context, cursor, true);
 				} else {
 					Log.e(TAG, "did not found conversation: " + threadId);
+				}
+				if (cursor != null && !cursor.isClosed()) {
+					cursor.close();
 				}
 			}
 			return ret;
