--- BuggyCode.java (Com Erro)
+++ GroundTruth_Fix.java (Corrigido)
@@ -110,7 +110,7 @@
 	private Timer shutdownTimer = null;
 	private Timer idleTimer = null;
 	public long idle_interval = 1000;
-    private int failCount = 0;
+    private volatile int failCount = 0;
 	private long reconnect_timestamp = 0;
 	private String useragent = null;
     private String streamId = null;
@@ -606,6 +606,9 @@
             Log.d(TAG, "Connecting: " + url);
         }
 
+        TestFlight.log("Attempt: " + failCount);
+        Log.d(TAG, "Attempt: " + failCount);
+
 		client = new WebSocketClient(URI.create(url), new WebSocketClient.Listener() {
 		    @Override
 		    public void onConnect() {
@@ -1703,6 +1706,10 @@
 			if(idleTimer != null && state != STATE_CONNECTED) {
 				idleTimer.cancel();
 				idleTimer = null;
+                failCount = 0;
+                if(wifiLock.isHeld())
+                    wifiLock.release();
+                reconnect_timestamp = 0;
 				state = STATE_DISCONNECTED;
 			}
 		}
