--- BuggyCode.java (Com Erro)
+++ GroundTruth_Fix.java (Corrigido)
@@ -206,19 +206,21 @@
     }
 
     private Bitmap getPreviewIcon() {
+        Bitmap icon = null;
         try {
-            return BitmapFactory.decodeStream(
-                       mContext.getContentResolver().openInputStream(
+            InputStream input = mContext.getContentResolver().openInputStream(
                            AttachmentProvider.getAttachmentThumbnailUri(mAccount,
                                    part.getAttachmentId(),
                                    62,
-                                   62)));
+                                   62));
+            icon = BitmapFactory.decodeStream(input);
+            input.close();
         } catch (Exception e) {
             /*
              * We don't care what happened, we just return null for the preview icon.
              */
-            return null;
-        }
+        }
+        return icon;
     }
 
     private void onViewButtonClicked() {
