--- BuggyCode.java (Com Erro)
+++ GroundTruth_Fix.java (Corrigido)
@@ -66,20 +66,23 @@
 
         File tempFile = new File(mContext.getCacheDir(), mFilename);
 
+
         DataOutputStream oos = new DataOutputStream(new FileOutputStream(tempFile));
 
-        oos.writeInt(numEntries);
-
-        while (it.hasNext()) {
-            Parcel p = Parcel.obtain(); // creating empty parcel object
-            p.writeParcelable(it.next(), 0); // saving bundle as parcel
-            byte[] buf = p.marshall();
-            oos.writeInt(buf.length);
-            oos.write(buf);
-            p.recycle();
-        }
-
-        oos.close();
+        try {
+            oos.writeInt(numEntries);
+
+            while (it.hasNext()) {
+                Parcel p = Parcel.obtain(); // creating empty parcel object
+                p.writeParcelable(it.next(), 0); // saving bundle as parcel
+                byte[] buf = p.marshall();
+                oos.writeInt(buf.length);
+                oos.write(buf);
+                p.recycle();
+            }
+        } finally {
+            oos.close();
+        }
 
     }
 
