--- BuggyCode.java (Com Erro)
+++ GroundTruth_Fix.java (Corrigido)
@@ -840,12 +840,19 @@
                             String sortOrder) throws Exception {
             mSemaphore.acquire();
 
-            final Cursor cursor;
-            cursor = mDelegate.query(uri, projection, selection, selectionArgs, sortOrder);
+            Cursor cursor = null;
+            try {
+                cursor = mDelegate.query(uri, projection, selection, selectionArgs, sortOrder);
+            } finally {
+                if (cursor == null) {
+                    mSemaphore.release();
+                }
+            }
 
             /* Android content resolvers can only process CrossProcessCursor instances */
             if (!(cursor instanceof CrossProcessCursor)) {
                 Log.w(K9.LOG_TAG, "Unsupported cursor, returning null: " + cursor);
+                mSemaphore.release();
                 return null;
             }
 
